File: frontend/src/types/job.ts
import type { IImage } from './image'

export interface ISkillReference {
  id: string
  name: string
  rating: number
  image?: IImage | null
  skillGroupId: string
}

export interface IJob {
  id: string
  title: string
  company: string
  companyUrl?: string
  companyImage?: IImage | null
  startDate: string
  endDate?: string | null
  location: string
  shortDescription: string
  isEducation: boolean
  includeOnResume: boolean
}

export interface IJobDetail extends IJob {
  longDescription: string
  skills: ISkillReference[]
}

File: backend/src/main/java/com/simonrowe/employment/Job.java
package com.simonrowe.employment;

import com.simonrowe.common.Image;
import java.util.List;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "jobs")
public record Job(
    @Id String id,
    String title,
    String company,
    String companyUrl,
    Image companyImage,
    String startDate,
    String endDate,
    String location,
    String shortDescription,
    String longDescription,
    Boolean isEducation,
    Boolean includeOnResume,
    List<String> skills
) {
}
